react/bootstrap/src/components/tree/tree.tsx

```diff
@@ -0,0 +1,99 @@
+import {Slot} from '@agnos-ui/react-headless/slot';
+import {useDirective} from '@agnos-ui/react-headless/utils/directive';
+import {useWidgetWithConfig} from '../../config';
+import type {TreeContext, TreeProps, TreeSlotItemContext} from './tree.gen';
+import {createTree} from './tree.gen';
+import classNames from 'classnames';
+
+/**
+ * A functional component that renders a toggle element with a directive applied to it.
+ * The directive is provided through the `slotContext` parameter.
+ *
+ * @param slotContext - The context object containing the directives and item the toggle.
+ * @returns A toggle element with the applied directive.
+ */
+export const DefaultTreeSlotToggle = (slotContext: TreeSlotItemContext) => {
+	const {directives, item} = slotContext;
+	return item.children!.length > 0 ? (
+		<button {...useDirective(directives.itemToggleDirective, {item})}></button>
```

Comment:
`useDirective` should always at the top of a react component and not in conditions ( as described in React's [Rules of Hooks](https://react.dev/reference/rules/rules-of-hooks))
note that [this PR](https://github.com/AmadeusITGroup/AgnosUI/pull/1025) will avoid to make this kind of mistakes through eslint rules